### ISSUE TITLE
On-the-fly patchbay activation from command line

### ISSUE THREAD

**Issue by fpesari** [rncbc/qjackctl#67]
The patchbay is very useful for live performance, however patchbay profiles can be activated only via the GUI, which makes it hard to automate via scripting.

It would be nice if QjackCtl offered a way to active patchbays on the fly, without restarting the JACK server.

I'm not asking to change the current `-a` option if that means breaking some people's current setups, but maybe a new option like this: 

`qjackctl --patchbay-profile=Song1.xml`

would be very, very useful to all people who make their own live setups without relying on DAWs. 

Thank you for reading :smile_cat:

**Comment by rncbc** [comment]
as you might know qjackctl is a GUI interface and it will be always like so... forever :) actually you don't need to restart the jack-server to switch over patchbay profiles: you just select/load one and activate it, but yes only from the GUI...
as better suited for the command-line and/or scripting there are alternatives out there like jack.plumbing, perhaps...
cheers

**Comment by fpesari** [comment]
> as you might know qjackctl is a GUI interface and it will be always like so... forever :) actually you don't need to restart the jack-server to switch over patchbay profiles: you just select/load one and activate it, but yes only from the GUI...
> as better suited for the command-line and/or scripting there are alternatives out there like jack.plumbing, perhaps...
> cheers

Yes, I know qjackctl is a graphical program, but this feature would only use things which already present, especially the comfortable and already widespread Patchbay XML format. But if it's troublesome to implement, I understand!

**Comment by rncbc** [comment]
> Yes, I know qjackctl is a graphical program, but this feature would only use things which already present, especially the comfortable and already widespread Patchbay XML format. But if it's troublesome to implement, I understand!

nope, not that troublesome, in fact it may-be feasible from the command line alright--assuming the single/unique instance option is set and/or through the native d-bus interface as a new method (eg. "active-patchbay" with the path to the xml profile as a string argument perhaps).

**Comment by rncbc** [comment]
hopefully implemented in https://github.com/rncbc/qjackctl/commit/e76e58e

please check it out, test && tell
cheers

**Comment by fpesari** [comment]
Tested on Debian, works just fine. Thank you!!!

**Comment by binjch** [comment]
Sorry, i'm just a simple musician but i'll be glad to enjoy your trick :-) How to do it practicly (if you explain how to read and  how to enforce the informations on the present page it's even better, so i learn)? I also would like to activate the patchbay at session startup. Thanks!

**Comment by rncbc** [comment]
from the command line, works even if it's already running:
`qjackctl --active-patchbay=/path/to/patchbay-profile.xml`
